File: src/main/java/uk/co/lukestevens/testing/mocks/MockEncryptionService.java
package uk.co.lukestevens.testing.mocks;

import java.io.IOException;

import uk.co.lukestevens.encryption.EncryptionService;

public class MockEncryptionService implements EncryptionService {
	
	private static final String SUFFIX = " (encrypted)";

	@Override
	public String encrypt(String plaintext) throws IOException {
		return plaintext + SUFFIX;
	}

	@Override
	public String decrypt(String ciphertext) throws IOException {
		return ciphertext.substring(0, ciphertext.length() - SUFFIX.length());
	}

}

File: src/main/kotlin/uk/co/lukestevens/testing/mocks/MockEncryptionService.kt
package uk.co.lukestevens.testing.mocks

import java.io.IOException
import uk.co.lukestevens.encryption.EncryptionService

class MockEncryptionService : EncryptionService {
	
	override fun encrypt(plaintext: String): String {
		return plaintext + SUFFIX
	}

	override fun decrypt(ciphertext: String): String {
		return ciphertext.substring(0, ciphertext.length - SUFFIX.length)
	}

	companion object {
		private val SUFFIX = " (encrypted)"
	}
}